consensus/istanbul/backend/engine.go

```diff
@@ -921,10 +921,69 @@ func (sb *backend) snapshot(chain consensus.ChainReader, number uint64, hash com
 		logger.Trace("Stored voting snapshot to disk", "number", snap.Number, "hash", snap.Hash)
 	}
 
+	sb.regen(chain, headers)
+
 	sb.recents.Add(snap.Hash, snap)
 	return snap, err
 }
 
+// regen commits snapshot data to database
+// regen is triggered if there is any checkpoint block in the `headers`.
+// For each checkpoint block, this function verifies the existence of its snapshot in DB and stores one if missing.
+/*
+ Triggered:
+ |   ^                          ^                          ^                          ^  ...|
+     SI                 SI*(last snapshot)                 SI                         SI
+       			   | header1, .. headerN |
+ Not triggered: (Guaranteed SI* was committed before )
+ |   ^                          ^                          ^                          ^  ...|
+     SI                 SI*(last snapshot)                 SI                         SI
+	                            | header1, .. headerN |
+*/
+func (sb *backend) regen(chain consensus.ChainReader, headers []*types.Header) {
+	if !sb.isRestoringSnapshots.Load() && len(headers) > 1 {
+		var (
+			from        = headers[0].Number.Uint64()
+			to          = headers[len(headers)-1].Number.Uint64()
+			start       = time.Now()
+			commitTried = false
+		)
+
+		// Shortcut: No missing snapshot data to be processed.
+		if to-(to%uint64(params.CheckpointInterval)) < from {
+			return
+		}
+
+		sb.isRestoringSnapshots.Store(true)
+		defer func() {
+			sb.isRestoringSnapshots.Store(false)
+		}()
```

Comment:
It would be better to print a log message informing the start of snapshot restoring

Reply:
Considering entry message should not be printed excessively, that message also should be printed [under the condition](https://github.com/klaytn/klaytn/pull/2074/files#diff-44abfea3bdbe16ce301a955732b485d1f0af2cc6c0c4cee3fbce0feb6ce051b0R981). Putting the initiating message seems redundant and not effective under this condition.